src/app/stats/stats.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, Input, OnInit, OnChanges } from '@angular/core';
+import _ from 'lodash';
+import { Stats } from './stats';
+
+import { Todo } from './../todo';
+
+@Component({
+  selector: 'stats' ,
+  templateUrl: './stats.component.html'
+})
+
+export class StatsComponent implements OnInit, OnChanges{
+  @Input('init') todos: Todo[]=[];
+  stats: Stats;
+  ngOnInit(){
+    this.updateStats(this.todos);
+  }
+
+  ngOnChanges(changes){
+    this.updateStats(changes.todos.currentValue);
+  }
+  updateStats(todos){
+    let done = _.reduce(todos, (completed, todo: Todo)=>{
```

Comment:
make this updateStats code smaller. you can do this with only 1 loop why 2 loops?

Reply:
ok sir!